tests/providers/amazon/conftest.py

```diff
@@ -0,0 +1,59 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import warnings
+
+try:
+    import importlib_metadata
+except ImportError:
+    from importlib import metadata as importlib_metadata  # type: ignore[no-redef]
+
+import pytest
+
+
+@pytest.fixture(scope="session")
+def botocore_version():
+    try:
+        version = importlib_metadata.version("botocore")
+    except importlib_metadata.PackageNotFoundError:
+        warnings.warn("'botocore' package not found'", UserWarning)
+        return None
+
+    try:
+        return tuple(map(int, version.split(".")[:3]))
+    except Exception:
+        warnings.warn(f"Unable to parse botocore {version!r}", UserWarning)
+        return None
+
+
+@pytest.fixture(autouse=True)
+def filter_botocore_warnings(botocore_version):
+    """Filter known botocore future warnings."""
+
+    with warnings.catch_warnings():
+        if botocore_version and botocore_version < (1, 28):
+            # Deprecate Usage of sslComonName in Endpoint Creation.
+            # See: https://github.com/boto/botocore/issues/2705
+            warnings.filterwarnings(
+                "ignore",
+                category=FutureWarning,
+                module="botocore.client",
+                message="The .* client is currently using a deprecated endpoint.*",
+            )
+        yield
```

Comment:
Seems like latest botocore (at least 1.28.0) still show warning that legacy endpoints use even if should changed to new endpoints.
I will check is it reproduce in regular environment or it related to some `moto` behaviour

```
tests/providers/amazon/aws/hooks/test_emr.py::TestEmrHook::test_get_conn_returns_a_boto3_connection
    /usr/local/lib/python3.7/site-packages/botocore/client.py:627: FutureWarning: The elasticmapreduce client is currently using a deprecated endpoint: ap-southeast-2.elasticmapreduce.amazonaws.com. In the next minor version this will be moved to elasticmapreduce.ap-southeast-2.amazonaws.com. See https://github.com/boto/botocore/issues/2705 for more details.
      category=FutureWarning,
  
  tests/providers/amazon/aws/hooks/test_rds.py: 1 warning
  tests/providers/amazon/aws/operators/test_rds.py: 28 warnings
  tests/providers/amazon/aws/sensors/test_rds.py: 10 warnings
    /usr/local/lib/python3.7/site-packages/botocore/client.py:627: FutureWarning: The rds client is currently using a deprecated endpoint: rds.amazonaws.com. In the next minor version this will be moved to rds.us-east-1.amazonaws.com. See https://github.com/boto/botocore/issues/2705 for more details.
      category=FutureWarning,
  
  tests/providers/amazon/aws/hooks/test_sqs.py::TestSqsHook::test_get_conn
  tests/providers/amazon/aws/operators/test_sqs.py::TestSqsPublishOperator::test_execute_failure_fifo_queue
  tests/providers/amazon/aws/operators/test_sqs.py::TestSqsPublishOperator::test_execute_success
  tests/providers/amazon/aws/operators/test_sqs.py::TestSqsPublishOperator::test_execute_success_fifo_queue
  tests/providers/amazon/aws/sensors/test_sqs.py::TestSqsSensor::test_poke_batch_messages
  tests/providers/amazon/aws/sensors/test_sqs.py::TestSqsSensor::test_poke_message_invalid_filtering
  tests/providers/amazon/aws/sensors/test_sqs.py::TestSqsSensor::test_poke_no_message_failed
  tests/providers/amazon/aws/sensors/test_sqs.py::TestSqsSensor::test_poke_success
    /usr/local/lib/python3.7/site-packages/botocore/client.py:627: FutureWarning: The sqs client is currently using a deprecated endpoint: queue.amazonaws.com. In the next minor version this will be moved to sqs.us-east-1.amazonaws.com. See https://github.com/boto/botocore/issues/2705 for more details.
      category=FutureWarning,
```

Reply:
Yeah, seems like even after upgrade to latest minor version of `botocore` and `boto3` it sill use `{region}.{service}.{dnsSuffix}` instead of `{service}.{region}.{dnsSuffix}`, as result this warnings appear in to the test logs.

I thought this is fine for regular DAG usage, users would informed that some changes might happen in the future.

But what we should do in tests?
- Remove fixtures from `conftest.py` and keep  this warnings in CI log
- Remove  upper bound  pin to specific botocore version

WDYT @potiuk @uranusjr @eladkal 

In additional I've asked in [botocore issue page](https://github.com/boto/botocore/issues/2705) about default behaviour.

Reply:
Got an answer that default endpoints will changed in botocore 1.29